lib/main.dart

```diff
@@ -1,78 +1,110 @@
+import 'package:meeting_scheduler/profile/screen_profile_details.dart';
+import 'package:meeting_scheduler/project/screen_project_details.dart';
+import 'package:meeting_scheduler/team/screen_team_details.dart';
 
 import 'firebase_options.dart';
 
 void main() {
-  runApp(const MyApp());
+  runApp(MyApp());
 }
 
 class MyApp extends StatelessWidget {
-  const MyApp({Key? key}) : super(key: key);
+  MyApp({Key? key}) : super(key: key);
 
   // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
     return FutureBuilder(
-        future: Firebase.initializeApp(
-            options: DefaultFirebaseOptions.currentPlatform),
-        builder: (context, snapshot) {
-          if (snapshot.connectionState == ConnectionState.waiting) {
-            return const CircularProgressIndicator();
-          } else {
-            return MaterialApp(
-              title: 'Meeting Scheduler',
-              theme: ThemeData(
-                primarySwatch: Colors.blue,
-              ),
-              home: const AuthGate(),
-            );
-          }
-        });
-  }
-}
-
-class AuthGate extends StatelessWidget {
-  const AuthGate({Key? key}) : super(key: key);
-
-  @override
-  Widget build(BuildContext context) {
-    return StreamBuilder<User?>(
-      stream: FirebaseAuth.instance.authStateChanges(),
+      future: Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform),
       builder: (context, snapshot) {
-        return StreamBuilder<User?>(
-          stream: FirebaseAuth.instance.authStateChanges(),
-          builder: (context, snapshot) {
-            // User is not signed in
-            if (!snapshot.hasData) {
-              return const SignInScreen(providerConfigs: [
-                EmailProviderConfiguration(),
-                GoogleProviderConfiguration(
-                  clientId:
-                      '220727339122-b3fflgie9rq6b6atu8sfml3274snrs31.apps.googleusercontent.com',
-                ),
-              ]);
-            }
-
-            // Render your application if authenticated
-            return const HomePage();
-          },
-        );
+        if (snapshot.connectionState == ConnectionState.waiting) {
+          return const Center(
+            child: CircularProgressIndicator(),
+          );
+        } else {
+          return MaterialApp.router(
+            routeInformationParser: _router.routeInformationParser,
+            routerDelegate: _router.routerDelegate,
+            title: 'Meeting Scheduler',
+            theme: ThemeData(
+              primarySwatch: Colors.blue,
+            ),
+          );
+        }
       },
     );
   }
-}
-
-class HomePage extends StatelessWidget {
-  const HomePage({Key? key}) : super(key: key);
 
-  @override
-  Widget build(BuildContext context) {
-    return const Material(
-        child: Center(
-      child: Text('Hello'),
-    ));
-  }
+  final GoRouter _router = GoRouter(
```

Comment:
Is this the guidance from go_router - keep _router as a top level variable in main.dart?  Just out of interest

Reply:
Yes, it is done like that on all of the documentation.

Reply:
> it is done like that on all of the documentation

Apparently not *all* documentation - the only place I looked was https://pub.dev/packages/go_router but there the `_router` object is a member of the App widget.

I hope it's clear I'm not interested in trying to be 'right' or make anyone else 'wrong', when we have a difference in perspective it's an opportunity to get closer to the truth!

Personally I'm interested in this question because I don't know go_router at all and would like to understand if there are best practices or advantages/disadvantages to different architectures.

Reply:
Oh it's a member of HomePage?  I need get back to good habits and review in VS Code as well - github UI is good for commenting but reading code not so good imo... will do that now...

Reply:
Anyway, sorry - this is probably better to take to Discord rather than take up energy in the PR... please feel free to resolve 👍

Reply:
I see what you mean, the thing is I put the variable _router at the same level as build, that is how it is done on the docs, the thing is the MaterialApp is on one level under because of the future builder. I think the way it is now won't cause any problems. If you think _router should be at the same level of the MateralApp, it can be changed to that as well.

Reply:
I was just hoping to discuss in order to better understand as I'm new to go_router.  I wasn't suggesting that I thought there was a problem or that something needed to change.

Reply:
Oh just for clarity - I initially asked if keeping the _router object as a top level variable in `main.dart`, but it never was a top level variable in `main.dart` (that was my misunderstanding from not reading the code properly). So when you responded that all the docs say so and I disagreed, it was based on that initial misunderstanding.  Sorry for the confusion.